src/codeu/chat/client/core/View.java

```diff
@@ -68,6 +69,23 @@ public View(ConnectionSource source) {
     return users;
   }
 
+  public ServerInfo getInfo() {
+    try (final Connection connection = this.source.connect()) {
+      Serializers.INTEGER.write(connection.out(), NetworkCode.SERVER_INFO_REQUEST);
+      if (Serializers.INTEGER.read(connection.in()) == NetworkCode.SERVER_INFO_RESPONSE) {
+        final Uuid version = Uuid.SERIALIZER.read(connection.in());
+        return new ServerInfo(version);
```

Comment:
This doesn't look like the code in the tutorial - this will do a version check, but doesn't make use of your timing code, please take another look at the guide on page 7 :)